### DISCUSSION TITLE
Default Kingdom Buildings

### DISCUSSION THREAD

**Discussion by WhatAUsernameIHave** [Deijin27/RanseiLink#50] [General]
I decided to look into this and compile a list of all building defaults and what slot they go in. Also, Unknown 2 of Scenario Building seems to deal with if the Building is available when starting a story (1 meaning yes with any other number meaning no). Anything in ( ) is the number attached to the building in RanseiLink. Also, setting default dummy slots to 1 will not cause anything to show up.

Aurora:
Slot 0 - Shop (000)
Slot 1 - Ponigiri Shop (001)
Slot 2 - Bank (002)
Slot 3 - Blacksmith (003)
Slot 4 - Farm (004)
Slot 5 - Ravine (005)
Slot 6 - Sacred Ruins (006)

Ignis:
Slot 0 - Shop (007)
Slot 1 - Ponigiri Shop (008)
Slot 2 - Bank (009)
Slot 3 - Mystery Springs (010)
Slot 4 - Cave (011)
Slot 5 - Floating Rock (012)
Slot 6 - Shine of Truth (013)

Fontaine:
Slot 0 - Shop (014)
Slot 1 - Ponigiri Shop (015)
Slot 2 - Bank (016)
Slot 3 - River of Gold (017)
Slot 4 - Ravine (018)
Slot 5 - Farm (019)
Slot 6 - dummy_0 (020)

Violight:
Slot 0 - Shop (021)
Slot 1 - Ponigiri Shop (022)
Slot 2 - Bank (023)
Slot 3 - Power Plant (024)
Slot 4 - Sky Garden (025)
Slot 5 - Floating Rock (026)
Slot 6 - IIdyllic Skies (027)

Greenleaf:
Slot 0 - Ponigiri Shop (028)
Slot 1 - Gold Mine (029)
Slot 2 - Bank (030)
Slot 3 - Woodlands (031)
Slot 4 - Ravine (032)
Slot 5 - Farm (033)
Slot 6 - dummy_1 (034)

Nixtorm:
Slot 0 - Shop (035)
Slot 1 - Gold Mine (036)
Slot 2 - Bank (037)
Slot 3 - Blacksmith (038)
Slot 4 - Snowy Mountain (039)
Slot 5 - Sky Garden (040)
Slot 6 - Icy Peak (041)

Pugilis:
Slot 0 - Shop (042)
Slot 1 - Gold Mine (043)
Slot 2 - Bank (044)
Slot 3 - Arena (045)
Slot 4 - Park (046)
Slot 5 - Farm (047)
Slot 6 - dummy_2 (048)

Viperia:
Slot 0 - Shop (049)
Slot 1 - Gold Mine (050)
Slot 2 - Bank (051)
Slot 3 - Alchemist (052)
Slot 4 - Cave (053)
Slot 5 - Park (054)
Slot 6 - dummy_3 (055)

Terrera:
Slot 0 - Shop (056)
Slot 1 - Ponigiri Shop (057)
Slot 2 - Bank (058)
Slot 3 - Open-Pit Mine (059)
Slot 4 - Farm (060)
Slot 5 - Park (061)
Slot 6 - Ancient Cave (062)

Avia:
Slot 0 - Shop (063)
Slot 1 - Ponigiri Shop (064)
Slot 2 - Bank (065)
Slot 3 - Windmill (066)
Slot 4 - Sky Garden (067)
Slot 5 - Snowy Mountain (068)
Slot 6 - dummy_4 (069)

Illusio:
Slot 0 - Ponigiri Shop (070)
Slot 1 - Gold Mine (071)
Slot 2 - Bank (072)
Slot 3 - Power Spot (073)
Slot 4 - Floating Rock (074)
Slot 5 - Snowy Mountain (075)
Slot 6 - Mystery Rock (076)

Chrysalia:
Slot 0 - Shop (077)
Slot 1 - Gold Mine (078)
Slot 2 - Bank (079)
Slot 3 - Alchemist (080)
Slot 4 - Ravine (081)
Slot 5 - Sky Garden (082)
Slot 6 - dummy_5 (083)

Cragspur:
Slot 0 - Shop (084)
Slot 1 - Gold Mine (085)
Slot 2 - Bank (086)
Slot 3 - Hermitage (087)
Slot 4 - Cave (088)
Slot 5 - Park (089)
Slot 6 - dummy_6 (090)

Spectra:
Slot 0 - Ponigiri Shop (091)
Slot 1 - Gold Mine (092)
Slot 2 - Bank (093)
Slot 3 - Power Spot (094)
Slot 4 - Floating Rock (095)
Slot 5 - Cave (096)
Slot 6 - dummy_7 (097)

Dragnor:
Slot 0 - Shop (098)
Slot 1 - Gold Mine (099)
Slot 2 - Bank (100)
Slot 3 - Dragon Chamber (101)
Slot 4 - Sky Garden (102)
Slot 5 - Ravine (103)
Slot 6 - Infinite Tower (104)

Yaksha:
Slot 0 - Shop (105)
Slot 1 - Ponigiri Shop (106)
Slot 2 - Bank (107)
Slot 3 - Underground Mine (108)
Slot 4 - Cave (109)
Slot 5 - Floating Rock (110)
Slot 6 - dummy_8 (111)

Valora:
Slot 0 - Ponigiri Shop (112)
Slot 1 - Gold Mine (113)
Slot 2 - Bank (114)
Slot 3 - Mystery Spring (115)
Slot 4 - Snowy Mountain (116)
Slot 5 - Park (117)
Slot 6 - Shrine of Resolve (118)

**Comment by Deijin27** [comment]
That's awesome. I'll have a proper look through at some point if this corresponds to updates to be made to ranseilink.

I really like how the scnario warrior workspace came out, and may do a similar thing for buildings, visually organising them in a way that you can easily see the kingdom slots and move between.

What I was working on next was the animated images related to kingdoms, so people can flesh out custom kingdoms more.

But I got a bit distracted with the new Avalonia release v11. I'm impressed, and have already replicated most of the visual style of ranseilink over there. Theres a couple of points of difficulty still, but I really think this is a viable replacement for WPF at this point (considering I program with wpf every day, I need something similar to not be pointlessly handicapping development). RanseiLink may well be coming to mac and linux soon.
